src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -2670,6 +2681,12 @@ components:
           $ref: '#/components/schemas/UniqueIdProperty'
         schema:
           $ref: '#/components/schemas/DatasetSpecificationModel'
+        region:
+          type: String
+          default: us-central1
```

Comment:
Yeah, to your last point - I was thinking they were avoiding using "GoogleRegion" in order to support other future cloud platforms, like "AzureRegion". Setting this to be a string allows either GoogleRegion or AzureRegion to be valid, but then (I think) we're missing the benefits of using the open-api ref type, like input validation.

Reply:
That's a good point! Right now if a user enters an invalid region it uses "us-central1" instead of throwing a validation error, which is definitely confusing.

Reply:
We've updated the code so that if an invalid region is provided, or a region that doesn't exist in the specified cloud platform is provided, the user will get a 400 BAD_REQUEST. 

We don't have a good way right now, using OpenAPI, to support conditional values in the swagger page. I think this merits some further discussion and we should have a quick chat about it.